include/API/Device.h

```diff
@@ -46,7 +51,7 @@ class Device {
   llvm::StringRef getDescription() const { return Description; }
 
   static void registerDevice(std::shared_ptr<Device> D);
-  static llvm::Error initialize();
+  static llvm::Error initialize(const DeviceConfig Config);
```

Comment:
Isn't this `const` redundant / ignored?

Reply:
You can modify a pass-by-value variable, it just modifies the instance.